File: Kubernetes/SportIBook/backend/isportbook_main/models/feedback.py
from django.db import models
from .models_utils.integer_range_field import IntegerRangeField
from django.utils.translation import gettext_lazy as _

class Feedback(models.Model):
    min_value = 0
    max_value = 10
    cleaning = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Cleaning"))
    motivation = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Motivation"))
    value_for_money = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Value for money"))
    professionality = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Professionality"))
    passion = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Passion"))
    skill = IntegerRangeField(min_value=min_value, max_value=max_value, default=0, verbose_name=_("Skill"))
    booking = models.OneToOneField("isportbook_main.Booking", on_delete=models.CASCADE, verbose_name=_("Booking"))

    def __str__(self):
        return self.booking.__str__()

    class Meta:
        verbose_name = _("Feedback")
        verbose_name_plural = _("Feedbacks")

File: Kubernetes/05-SportIBook/frontend/src/app/models/feedback.ts
/* { "uuid": "bba5fa3b-e6d0-4d40-87dc-43350d6e3cc4", "pk": 8, "email": "alessandro@madfarm.it", "feedback": { "closeness":0, "punctuality":0, "value_for_money": 0, "skill": 0, "cleaning": 0, "passion": 0, "booking": 8}} */

export class Feedback {
    cleaning:  number;
    motivation: number;
    value_for_money: number;
    professionality: number;
    passion: number;
    skill: number;
    booking: number;
    constructor(cleaning: number, motivation: number, value_for_money: number, professionality: number, passion:number, skill:number, booking:number) {
        this.cleaning = cleaning;
        this.motivation = motivation;
        this.value_for_money = value_for_money;
        this.professionality = professionality;
        this.passion = passion;
        this.skill = skill;
        this.booking = booking;
    }
}
